AFQ/data.py

```diff
@@ -286,6 +293,1008 @@ def read_templates(resample_to=False):
+            deriv_types = []
+            for sub_key in deriv_subs:
+                deriv_types += [
+                    s.split(sub_key)[-1].lstrip('/')
+                    for s in _ls_s3fs(
+                        s3_prefix=sub_key,
+                        anon=self.anon
+                    )['subjects']
+                ]
+
+            return list(set(deriv_types))
+        else:
+            return []
+
+    def _get_non_subject_s3_keys(self):
+        """Return a list of 'non-subject' files
+
+        In this context, a 'non-subject' file is any file
+        or directory that is not a subject ID folder. This method
+        is different from AFQ.data.S3BIDSStudy because the HBN
+        dataset is not BIDS compliant
+
+        Returns
+        -------
+        dict
+            dict with keys 'raw' and 'derivatives' and whose values
+            are lists of S3 keys for non-subject files
+
+        See Also
+        --------
+        AFQ.data.S3BIDSStudy._get_non_subject_s3_keys
+        """
+        non_sub_s3_keys = {}
+
+        s3_prefix = '/'.join([self.bucket, self.s3_prefix])
+
+        nonsub_keys = _ls_s3fs(s3_prefix=s3_prefix,
+                               anon=self.anon)['other']
+        nonsub_keys = [k for k in nonsub_keys
+                       if not k.endswith('derivatives')]
+
+        nonsub_deriv_keys = _ls_s3fs(
+            s3_prefix='/'.join([
+                self.bucket,
+                self.s3_prefix,
+                'derivatives'
+            ]),
+            anon=self.anon
+        )['other']
+
+        non_sub_s3_keys = {
+            'raw': nonsub_keys,
+            'derivatives': nonsub_deriv_keys,
+        }
+
+        return non_sub_s3_keys
+
+    def download(self, directory, include_non_sub_raw_keys=False,
+                 include_derivs=False, overwrite=False, pbar=True):
+        """Download files for each subject in the study
+
+        Parameters
+        ----------
+        directory : str
+            Directory to which to download subject files
+
+        include_derivs : bool or str
+            If True, download all derivatives files. If False, do not.
+            If a string or sequence of strings is passed, this will
+            only download derivatives that match the string(s) (e.g.
+            ["dmriprep", "afq"]). Default: False
+
+        overwrite : bool
+            If True, overwrite files for each subject. Default: False
+
+        pbar : bool
+            If True, include progress bar. Default: True
+
+        See Also
+        --------
+        AFQ.data.S3BIDSSubject.download
+        """
+        super().download(
+            directory=directory,
+            include_non_sub_raw_keys=include_non_sub_raw_keys,
+            include_derivs=include_derivs,
+            overwrite=overwrite,
+            pbar=pbar
+        )
+
+        to_bids_description(
+            directory,
+            **{"BIDSVersion": "1.0.0",
+               "Name": "HBN Study, " + self.site})
```

Comment:
We might want to add some additional information here. For example, what subjects were selected this time. I think you can add a list here. (and yes, we should probably also do this for the HCP data as well, if we aren't).

Reply:
Thanks. Updated in latest commits.